akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -2226,8 +2226,10 @@ private[stream] object Collect {
               subOutlet.push(grab(in))
             }
             override def onUpstreamFinish(): Unit = {
-              subOutlet.complete()
-              maybeCompleteStage()
+              if (firstElementPushed) {
+                subOutlet.complete()
+                maybeCompleteStage()
+              }
```

Comment:
So if upstream finishes  here and we have not yet seen a pull from the inner flow we cancel out the finish here and then complete when after pushing the element on first pull, after emitting the element. Makes sense. 👍

Reply:
@johanandren , seems like reimplementing 'prepend' no?
what's wrong with leaving the prepend in place and completing the subOutlet anyway? this is the approach taken in `flatmapPrefix` where it's possible for the external graph stage to complete while the internal flow keeps running.

Reply:
Sounds like a good idea, I didn't consider that.